src/DataSource.ts

```diff
@@ -276,7 +277,12 @@ function parseServiceMapResponse(
   query?: XrayQuery
 ): DataFrame[] {
   const [servicesFrame, edgesFrame] = parseGraphResponse(response, query);
-  const serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subTitle}"))';
+  let serviceQuery;
+  if (NodeGraphDataFrameFieldNames.subTitle === 'subtitle') {
+    serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subtitle}"))';
+  } else {
+    serviceQuery = 'service(id(name: "${__data.fields.title}", type: "${__data.fields.subTitle}"))';
+  }
```

Comment:
```suggestion
const serviceQuery = `service(id(name: "\${__data.fields.title}", type: "\${__data.fields.${NodeGraphDataFrameFieldNames.subTitle}}"))`;
```

(I think this also works)

Reply:
This is definitely a cleaner approach and I tried something like this although I wasn't able to confirm this works across the different versions given my issues with getting the plugin to build.

Are you able to test it with this change?

Reply:
Great, thanks for checking 🙏